### ISSUE TITLE
Variable Stars

### ISSUE THREAD

**Issue by WhirligigGirl** [StollD/KopernicusExpansion-Continued#5]
Add an option to vary star intensity and color with time in curves, so that you would be able to produce variable stars including flare stars, cepheid variables, and solar weather cycles like our sun's.

Proposed syntax would look something like: 

```
ScaledVersion
{
	Variable //(or Light)
	{
		variabilityPeriodMin = //period of repeating curve in seconds, minimum value.
		variabilityPeriodMax = //period of repeating curve in seconds, maximum value.
                //for predictable stars, these should be about the same. But some stars may have more unpredictable variability periods that are not precise.
		intensityRandomness = //Error bars in percent of intensity at each point in the curve. High numbers may mean the intensity is fairly unpredictable over each cycle.
		//Randomness can also be used to create a simple unpredictable variable star with no curve.
		variableLensFlare = //True/False, if True, then the sunAU value will also be multiplied by the VariabilityCurve
                waitingPeriodMin = Minimum wait time of normal luminosity before starting the next variability cycle.
                waitingPeriodMax = Max wait time of normal luminosity before starting the next variability cycle.
                //The above two would be used for flare stars, which have a fairly predictable short period light curve for each individual flare, but for which the actual time between flares is fairly unpredictable.

		VariabilityCurve //each point multiplies the sun intensity and the solar panel power by the second value. (And the lens flare sunAU)
		{
			//key = timestamp where 1 = variability period, intensity value, 1st derivative, 2nd derivative.
			key = 0.0 1.0 0 0
			key = 0.4 0.9 0 0
			key = 1.0 1.0 0 0
		}
		ColorCurve //I used the Gradient atmo color rim syntax, but if there's a way to do it with the keys and the derivatives then that'd be ideal.
		{
			0.0 = RGBA(1,1,1,1)
			0.4 = RGBA(1,0.95,0.9,1)
			1.0 = RGBA(1,1,1,1)
		}
                //and as a stretch goal:
		RadiusCurve //Changes the radius (but NOT the mass) of the star in real time. Can be just the scaledspace model instead of the actual radius if that would cause problems.
		{
			key = 0.0 0.95 0 0
			key = 0.4 0.1 0 0
			key = 1.0 0.0.95 0 0
		}
	}
}
```
